nifi-nar-bundles/nifi-standard-bundle/nifi-standard-reporting-tasks/src/main/java/org/apache/nifi/controller/ControllerStatusReportingTask.java

```diff
@@ -117,25 +161,76 @@ public void onTrigger(final ReportingContext context) {
         builder.append(processorBorderLine);
         builder.append("\n");
 
-        printProcessorStatus(controllerStatus, builder, showDeltas);
+        printProcessorStatus(controllerStatus, builder, showDeltas, divisor);
 
         builder.append(processorBorderLine);
         processorLogger.info(builder.toString());
+    }
+
+    private void printConnectionStatuses(final ProcessGroupStatus controllerStatus, final boolean showDeltas, final long divisor) {
+        final StringBuilder builder = new StringBuilder();
 
-        builder.setLength(0);
         builder.append("Connection Statuses:\n");
         builder.append(connectionBorderLine);
         builder.append("\n");
         builder.append(connectionHeader);
         builder.append(connectionBorderLine);
         builder.append("\n");
 
-        printConnectionStatus(controllerStatus, builder, showDeltas);
+        printConnectionStatus(controllerStatus, builder, showDeltas, divisor);
 
         builder.append(connectionBorderLine);
         connectionLogger.info(builder.toString());
     }
 
+    private void printCounters(final ProcessGroupStatus controllerStatus, final boolean showDeltas, final long divisor) {
+        final StringBuilder builder = new StringBuilder();
+
+        builder.append("Counters:\n");
+        builder.append(counterBorderLine);
+        builder.append("\n");
+        builder.append(counterHeader);
+        builder.append(counterBorderLine);
+        builder.append("\n");
+
+        printCounterStatus(controllerStatus, builder, showDeltas, divisor);
+
+        builder.append(counterBorderLine);
+        counterLogger.info(builder.toString());
+    }
+
+    private void printCounterStatus(final ProcessGroupStatus status, final StringBuilder builder, final boolean showDeltas, final long divisor) {
+        final Collection<ProcessorStatus> processorStatuses = status.getProcessorStatus();
+        for (final ProcessorStatus processorStatus : processorStatuses) {
+            final Map<String, Long> counters = processorStatus.getCounters();
+
+            for (final Map.Entry<String, Long> entry : counters.entrySet()) {
+                final String counterName = entry.getKey();
+                final Long counterValue = entry.getValue() / divisor;
+
+                final String counterId = processorStatus.getId() + "_" + counterName;
+                final Long lastValue = lastCounterValues.getOrDefault(counterId, 0L);
+
+                lastCounterValues.put(counterId, counterValue);
+
+                if (showDeltas) {
+                    final String diff = toDiff(lastValue, counterValue);
+
+                    builder.append(String.format(COUNTER_LINE_FORMAT,
+                        processorStatus.getName() + "(" + processorStatus.getId() + ")",
```

Comment:
Since String.format() is being used, recommend incorporating the parentheses in the COUNTER_LINE_FORMAT definition to avoid the String concatenation.